File: mobile/lib/models/photo_model.dart
class PhotoModel {
  final int id;
  final int sessionId;
  final int? filterId;
  final String imagePath;
  final int orderIndex;
  final double beautySmooth;
  final double beautyBrighten;
  final String? filterName;
  final String? filterType;

  PhotoModel({
    required this.id,
    required this.sessionId,
    this.filterId,
    required this.imagePath,
    this.orderIndex = 0,
    this.beautySmooth = 0,
    this.beautyBrighten = 0,
    this.filterName,
    this.filterType,
  });

  factory PhotoModel.fromJson(Map<String, dynamic> json) {
    return PhotoModel(
      id: json['id'],
      sessionId: json['session_id'],
      filterId: json['filter_id'],
      imagePath: json['image_path'],
      orderIndex: json['order_index'] ?? 0,
      beautySmooth: (json['beauty_smooth'] ?? 0).toDouble(),
      beautyBrighten: (json['beauty_brighten'] ?? 0).toDouble(),
      filterName: json['filter_name'],
      filterType: json['filter_type'],
    );
  }

  Map<String, dynamic> toJson() => {
        'filter_id': filterId,
        'image_path': imagePath,
        'order_index': orderIndex,
        'beauty_smooth': beautySmooth,
        'beauty_brighten': beautyBrighten,
      };
}

File: backend/src/models/photoModel.js
const db = require('../config/database');

const PhotoModel = {
  create({ session_id, filter_id, image_path, order_index, beauty_smooth, beauty_brighten }) {
    const info = db
      .prepare(
        `INSERT INTO photos (session_id, filter_id, image_path, order_index, beauty_smooth, beauty_brighten)
         VALUES (?, ?, ?, ?, ?, ?)`
      )
      .run(
        session_id,
        filter_id || null,
        image_path,
        order_index ?? 0,
        beauty_smooth ?? 0,
        beauty_brighten ?? 0
      );
    return this.findById(info.lastInsertRowid);
  },

  findById(id) {
    return db.prepare(`SELECT * FROM photos WHERE id = ?`).get(id);
  },

  findBySession(session_id) {
    return db
      .prepare(`SELECT * FROM photos WHERE session_id = ? ORDER BY order_index ASC`)
      .all(session_id);
  },

  update(id, { filter_id, order_index, beauty_smooth, beauty_brighten }) {
    db.prepare(
      `UPDATE photos SET
        filter_id = COALESCE(?, filter_id),
        order_index = COALESCE(?, order_index),
        beauty_smooth = COALESCE(?, beauty_smooth),
        beauty_brighten = COALESCE(?, beauty_brighten)
       WHERE id = ?`
    ).run(filter_id, order_index, beauty_smooth, beauty_brighten, id);
    return this.findById(id);
  },

  remove(id) {
    db.prepare(`DELETE FROM photos WHERE id = ?`).run(id);
    return true;
  },
};

module.exports = PhotoModel;
